### PR TITLE
Merge 2.15.x into 3.0.x

### PR DESCRIPTION
**Pull request by derrabus**
Fun with Psalm Baselines. 🎉

### CODE REVIEW

File: lib/Doctrine/ORM/Mapping/MappingException.php [hunk 1]
```diff
@@ -344,6 +344,19 @@ public static function duplicateDiscriminatorEntry(string $className, array $ent
...
+    // * @param class-string $childEntityClass
+    // */
+    //public static function missingInheritanceTypeDeclaration(string $rootEntityClass, string $childEntityClass): self
+    //{
+    //    return new self(sprintf(
+    //        "Entity class '%s' is a subclass of the root entity class '%s', but no inheritance mapping type was declared.",
+    //        $childEntityClass,
+    //        $rootEntityClass
+    //    ));
+    //}
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Mapping/MappingException.php:358]: ?

Comment (greg0ire) [inline lib/Doctrine/ORM/Mapping/MappingException.php]: Let's do the switch to the new code suggested by @mpdude since it's causing issues?

**Review by greg0ire — APPROVED** [review]
Or part of the switch, also works 👍

**Comment by derrabus** [comment]
Yeah, I only switch the piece of code that caused trouble, so I can leave all the fun to @mpdude.

**Comment by greg0ire** [comment]
@mpdude I think the right way to do this would be not to commit these commented out lines of code and instead create a draft PR for 3.0 and link to it… you don't even know whether the code you're committing is actually going to work unless you try it on 3.0.

**Comment by derrabus** [comment]
Let's remove the commented-out code from 2.15.x again?

**Comment by mpdude** [comment]
#10465 removes comments

**Pull request merged**